app/javascript/crayons/MarkdownToolbar/markdownSyntaxFormatters.js

```diff
@@ -65,7 +65,9 @@ const handleLinkFormattingForEmptyTextSelection = ({
   selectionEnd,
 }) => {
   const basicFormattingForEmptySelection = {
-    newTextAreaValue: `${textBeforeSelection}[](${URL_PLACEHOLDER_TEXT})${textAfterSelection}`,
+    editSelectionStart: selectionStart,
```

Comment:
The changes in this file are all largely the same - instead of returning the whole `newTextAreaValue`, we instead return where the edit should happen, and what it should be replaced with. A helper function was added (line 395) that can construct the entire 'new text area value' as before, to help with testing and also the case where `document.execCommand` fails or isn't supported